horace_core/sqw/PixelData/@PixelData/PixelData.m

```diff
@@ -554,19 +666,44 @@
             pix_idx_end = obj.num_pixels;
         end
 
-        obj.data = obj.f_accessor_.get_pix(pix_idx_start, pix_idx_end);
-        if obj.page_size == obj.num_pixels && obj.f_accessor_.is_activated()
-            % Close the file if all pixels have been read
-            obj.f_accessor_.deactivate();
+        obj.data_ = obj.f_accessor_.get_pix(pix_idx_start, pix_idx_end);
+        if obj.page_size == obj.num_pixels
+            % Delete accessor and close the file if all pixels have been read
+            obj.f_accessor_ = [];
         end
-        obj.pix_position_ = pix_idx_start;
     end
 
-    function obj = load_first_page_if_data_empty_(obj)
-        % Check if there's any data in the current page and load a page if not
-        if isempty(obj.data_) && ~isempty(obj.f_accessor_)
-            obj = obj.load_page_(1);
+    function obj = load_dirty_page_(obj, page_number)
+        % Load a page of data from a tmp file
+        raw_pix = obj.tmp_io_handler.load_page(page_number);
+        npix_cols = obj.PIXEL_BLOCK_COLS_;
+        obj.data_ = reshape(raw_pix, [npix_cols, numel(raw_pix)/npix_cols]);
```

Comment:
why is the reshape here, not in the io_handler? 

In the current API, `write_page` takes a `nx9` array argument and load_page returns a `1x9n` vector. That difference is likely to lead to errors

Reply:
The reshape is in the `PixelData` class because only the `PixelData` instance knows how many columns it has. The IO handler doesn't necessarily need to know that information, it just reads/writes bytes, and the PixelData instance is responsible for knowing how to interpret those bytes. `write_page` can take any shape data